va/caa.go

```diff
@@ -229,30 +241,51 @@ func (va *ValidationAuthorityImpl) validateCAASet(caaSet *CAASet, wildcard bool)
 	//
 	// Our CAA identity must be found in the chosen checkSet.
 	for _, caa := range records {
-		if extractIssuerDomain(caa) == va.issuerDomain {
-			va.stats.Inc("CAA.Authorized", 1)
-			return true, true
+		caaIssuerDomain, caaParameters := extractIssuerDomainAndParameters(caa)
+		if caaIssuerDomain != va.issuerDomain {
+			continue
+		}
+
+		if features.Enabled(features.CAAValidationMethods) {
+			caaMethods, ok := caaParameters["validation-methods"]
+			if ok {
+				if challengeType == nil {
+					continue
+				}
+				if !containsMethod(caaMethods, *challengeType) {
+					continue
+				}
+			}
```

Comment:
I think the above block would benefit from a short comment about the validation being done perhaps with a reference to the ACME-CAA draft section on "validation-methods".

Reply:
Done.